File: app/src/main/java/tvshow/TvShowMainAdapter.java
package tvshow;

import android.content.Context;
import android.content.Intent;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.ProgressBar;
import android.widget.TextView;

import androidx.annotation.Keep;
import androidx.annotation.NonNull;
import androidx.fragment.app.FragmentActivity;
import androidx.recyclerview.widget.RecyclerView;

import com.squareup.picasso.Callback;
import com.squareup.picasso.Picasso;

import br.com.icaro.filme.R;
import domain.ListaItemSerie;
import domain.ListaSeries;
import tvshow.activity.TvShowActivity;
import utils.Constantes;
import utils.UtilsApp;


/**
 * Created by icaro on 17/02/17.
 */
@Keep
public class TvShowMainAdapter extends RecyclerView.Adapter<TvShowMainAdapter.TvShowPopularesViewHolder>{
    private Context context;
    private ListaSeries popularTvshow;

    public TvShowMainAdapter(FragmentActivity activity, ListaSeries popularTvshow) {
        context = activity;
        this.popularTvshow = popularTvshow;
    }

    @Override
    public TvShowMainAdapter.TvShowPopularesViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(context).inflate(R.layout.poster_main, parent, false);
        return new TvShowPopularesViewHolder(view);
    }

    @Override
    @Keep
    public void onBindViewHolder(@NonNull final TvShowMainAdapter.TvShowPopularesViewHolder holder, final int position) {
        final ListaItemSerie series = popularTvshow.getResults().get(position);

        Picasso.get()
                .load(UtilsApp.getBaseUrlImagem( UtilsApp.getTamanhoDaImagem(context, 2)) + series.getPosterPath())
                .error(R.drawable.poster_empty)
                .into(holder.img_poster_grid, new Callback() {
                    @Override
                    public void onSuccess() {
                        holder.progress_poster_grid.setVisibility(View.GONE);
                    }

                    @Override
                    public void onError(Exception e) {
                        holder.progress_poster_grid.setVisibility(View.GONE);
                        holder.title_main.setText(series.getName());
                        holder.title_main.setVisibility(View.VISIBLE);
                    }

                });

        holder.img_poster_grid.setOnClickListener(view -> {

            Intent intent = new Intent(context, TvShowActivity.class);
            intent.putExtra(Constantes.INSTANCE.getNOME_TVSHOW(), series.getName());
            intent.putExtra(Constantes.INSTANCE.getTVSHOW_ID(), series.getId());
            intent.putExtra(Constantes.INSTANCE.getCOLOR_TOP(), UtilsApp.loadPalette(holder.img_poster_grid));
            context.startActivity(intent);
        });

    }

    @Override
    public int getItemCount() {
        if (popularTvshow != null){
            return  popularTvshow.getResults().size() < 15 ? popularTvshow.getResults().size() : 15;
        }
        return  0;
    }

    @Keep
    class TvShowPopularesViewHolder extends RecyclerView.ViewHolder {

       private TextView title_main;
        private ProgressBar progress_poster_grid;
        private ImageView img_poster_grid;


        TvShowPopularesViewHolder(View itemView) {
            super(itemView);
            title_main = (TextView) itemView.findViewById(R.id.title_main);
            progress_poster_grid = (ProgressBar) itemView.findViewById(R.id.progress_poster_grid);
            img_poster_grid = (ImageView) itemView.findViewById(R.id.img_poster_grid);

        }
    }

    @Override
    public long getItemId(int position) {
        return super.getItemId(position);
    }
}

File: app/src/main/java/tvshow/TvShowMainAdapter.kt
package tvshow

import android.content.Context
import android.content.Intent
import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.annotation.Keep
import androidx.core.app.ActivityCompat
import androidx.fragment.app.FragmentActivity
import androidx.recyclerview.widget.RecyclerView
import br.com.icaro.filme.R
import domain.ListaItemSerie
import domain.ListaSeries
import kotlinx.android.synthetic.main.poster_main.view.img_poster_grid
import kotlinx.android.synthetic.main.poster_main.view.layout_poster_main
import kotlinx.android.synthetic.main.poster_main.view.progress_poster_grid
import kotlinx.android.synthetic.main.poster_main.view.title_main
import tvshow.activity.TvShowActivity
import utils.Constant
import utils.UtilsApp
import utils.getNameTypeReel
import utils.gone
import utils.setPicassoWithCache
import utils.visible
import utils.yearDate

/**
 * Created by icaro on 17/02/17.
 */
@Keep
class TvShowMainAdapter(activity: FragmentActivity, private val popularTvshow: ListaSeries) : RecyclerView.Adapter<TvShowMainAdapter.TvShowPopularesViewHolder>() {
    private val context: Context
    private var color: Int = 0

    init {
        context = activity
        this.color = ActivityCompat.getColor(context, R.color.white)
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int) = TvShowPopularesViewHolder(parent)

    override fun onBindViewHolder(holder: TvShowPopularesViewHolder, position: Int) {
        holder.bind(popularTvshow.results[position])
    }

    override fun getItemCount() = if (popularTvshow.results.size < 15) popularTvshow.results.size else 15

    inner class TvShowPopularesViewHolder(parent: ViewGroup) :
        RecyclerView.ViewHolder(LayoutInflater.from(context).inflate(R.layout.poster_main, parent, false)) {
        fun bind(item: ListaItemSerie) = with(itemView) {
            img_poster_grid.setBackgroundColor(ActivityCompat.getColor(context, R.color.accent))
            img_poster_grid.setPicassoWithCache(item.posterPath, 6,
                {
                    progress_poster_grid.gone()
                    color = UtilsApp.loadPalette(img_poster_grid)
                    layout_poster_main.setCardBackgroundColor(color)
                },
                {
                    progress_poster_grid.gone()
                    title_main.apply {
                        text = item.name
                        title_main.visible()
                    }
                }, img_erro = R.drawable.poster_empty)

            img_poster_grid.setOnClickListener {
                context.startActivity(Intent(context, TvShowActivity::class.java).apply {
                    putExtra(Constant.NOME_TVSHOW, item.name)
                    putExtra(Constant.TVSHOW_ID, item.id)
                    putExtra(Constant.ID_REEL, "${item.originalName?.getNameTypeReel()}-${item.firstAirDate?.yearDate()}")
                    putExtra(Constant.COLOR_TOP, color)
                })
            }
        }
    }
}
